packages/teleport/src/Databases/AddDatabase/Manually/Manually.tsx

```diff
@@ -0,0 +1,162 @@
+          </Text>
+          {' - Login to Teleport'}
+          <TextSelectCopy
+            mt="2"
+            text={`tsh login --proxy=${host} --auth=${cfg.getAuthType()} --user=${user}`}
+          />
+        </Box>
+        <Box mb={4}>
+          <Text bold as="span">
+            Step 3
+          </Text>
+          {' - Generate a join token'}
+          <TextSelectCopy mt="2" text="tctl tokens add --type=db" />
+        </Box>
+        <Box mb={4}>
+          <Text bold as="span">
+            Step 4
+          </Text>
+          {` - Select the database type and protocol to use`}
+          <Box mt={2}>
+            <Select
+              value={selectedDbOption}
+              onChange={(o: Option<Database>) => setSelectedDpOption(o)}
+              options={dbOptions}
+              isSearchable={true}
+            />
+          </Box>
+        </Box>
+        <Box mb={4}>
+          <Text bold as="span">
+            Step 5
+          </Text>
+          {' - Start the Teleport agent with the following parameters'}
+          <TextSelectCopy
+            mt="2"
+            text={`${generateDbStartCmd(
+              selectedDbOption.value.type,
+              selectedDbOption.value.protocol,
+              host
+            )}`}
+          />
+        </Box>
+        <Box>
+          {`* Note: For a self-hosted Teleport version, you may need to update DNS and obtain a TLS certificate for this application.
+            Learn more about database access `}
+          <Link
+            href={'https://goteleport.com/docs/database-access/'}
+            target="_blank"
+            mr="2"
+          >
+            here
+          </Link>
+        </Box>
+      </DialogContent>
+      <DialogFooter>
+        <ButtonSecondary onClick={onClose}>Close</ButtonSecondary>
+      </DialogFooter>
+    </>
+  );
+}
+
+export const generateDbStartCmd = (
+  type: DbType,
+  protocol: DbProtocol,
+  host: string
+) => {
+  const baseCommand = `teleport start --roles=db --token=[generated-join-token] --auth-server=${host} --db-name=[db-name] --db-protocol=${protocol} --db-uri=[uri]`;
+
+  switch (type) {
+    case 'self-hosted':
+      return baseCommand;
+    case 'rds':
+      return `${baseCommand} --db-aws-region=[region]`;
+    case 'redshift':
+      return `${baseCommand} --db-aws-region=[region] --db-aws-redshift-cluster-id=[cluster-id]`;
+    case 'gcp':
+      return `${baseCommand} --db-ca-cert=[instance-ca-filepath] --db-gcp-project-id=[project-id] --db-gcp-instance-id=[instance-id]`;
+    default:
+      return 'Unknown type and protocol';
+  }
+};
+
+const options: Database[] = [
+  formatDatabaseInfo('rds', 'postgres'),
+  formatDatabaseInfo('rds', 'mysql'),
+  formatDatabaseInfo('redshift', 'postgres'),
+  formatDatabaseInfo('gcp', 'postgres'),
+  formatDatabaseInfo('self-hosted', 'postgres'),
+  formatDatabaseInfo('self-hosted', 'mysql'),
+];
+
+type Database = {
+  type: DbType;
+  protocol: DbProtocol;
+  title: string;
+};
```

Comment:
hm... isn't this just the return types of `formatDatabaseInfo`?

Reply:
It is, so I can use `ReturnType` instead of redefining the type